### PR TITLE
Add s3 lock service

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**

### CODE REVIEW

File: proto/kvrpcpb.proto [hunk 1]
```diff
@@ -1428,3 +1428,55 @@ message CompactResponse {
     bytes compacted_start_key = 3;
     bytes compacted_end_key = 4;
 }
+
+message S3LockError {
```
Comment (JaySon-Huang) [inline proto/kvrpcpb.proto:1432]: Move all these message into `disaggregated.proto`

Comment (Lloyd-Pottiger) [reply proto/kvrpcpb.proto:1432]: ok

Comment (Lloyd-Pottiger) [reply proto/kvrpcpb.proto]: done

**Review by JaySon-Huang — APPROVED** [review]

**Pull request merged**